### PR TITLE
fix: frequent flush cause minio rate limit 

### PR DESCRIPTION
**Pull request by xiaofan-luan**
related to #28549
pr: #28626

1. avoid duplicated sync segments under syncing states
2. add jitter to avoid sync segments at the same time

### CODE REVIEW

**Comment by bigsheeper** [comment]
Actually, there is a logic to avoid duplicated sync segments:
https://github.com/milvus-io/milvus/blob/0fbd610e89d0e00eaaa93973f122fb9962710294/internal/datanode/flow_graph_insert_buffer_node.go#L468

**Comment by XuanYang-cn** [comment]
> Actually, there is a logic to avoid duplicated sync segments:
> 
> https://github.com/milvus-io/milvus/blob/0fbd610e89d0e00eaaa93973f122fb9962710294/internal/datanode/flow_graph_insert_buffer_node.go#L468

@bigsheeper  This checks when executing tasks, which is too late. we're generating way too many sync tasks,  logging them and merging them, but not executing them, which's very confusing. It's betther we ignore the syncing segment while generating sync task, more efficient.

File: internal/datanode/flush_manager.go [hunk 1]
```diff
@@ -683,6 +683,7 @@ func (t *flushBufferInsertTask) flushInsertData() error {
 		err := group.Wait()
 		metrics.DataNodeSave2StorageLatency.WithLabelValues(fmt.Sprint(paramtable.GetNodeID()), metrics.InsertLabel).Observe(float64(tr.ElapseSpan().Milliseconds()))
 		if err == nil {
+			log.Warn("failed to flush insert data", zap.Error(err))
```
Comment (XuanYang-cn) [inline internal/datanode/flush_manager.go:686]: I don't think this is the right place to log failed to flush, its `err==nil` above

File: internal/datanode/flush_manager.go [hunk 2]
```diff
@@ -707,6 +708,7 @@ func (t *flushBufferDeleteTask) flushDeleteData() error {
 		metrics.DataNodeSave2StorageLatency.WithLabelValues(fmt.Sprint(paramtable.GetNodeID()), metrics.DeleteLabel).Observe(float64(tr.ElapseSpan().Milliseconds()))
 		if err == nil {
 			for _, d := range t.data {
+				log.Warn("failed to flush delete data", zap.Error(err))
```
Comment (XuanYang-cn) [inline internal/datanode/flush_manager.go:711]: ditto

**Review by XuanYang-cn — CHANGES_REQUESTED** [review]

**Pull request merged**